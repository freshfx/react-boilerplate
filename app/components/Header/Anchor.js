import styled from 'styled-components'
import NormalA from 'components/Anchor'

const Anchor = styled(NormalA)`
  padding: 2em 0;
`

export default Anchor
