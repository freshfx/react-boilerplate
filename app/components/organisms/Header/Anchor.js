import styled from '@emotion/styled'

import NormalA from 'components/atoms/Anchor'

const Anchor = styled(NormalA)`
  padding: 2em 0;
`

export default Anchor
