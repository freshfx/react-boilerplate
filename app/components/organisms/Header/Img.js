import styled from '@emotion/styled'

import NormalImg from 'components/atoms/Img'

const Img = styled(NormalImg)`
  width: 100%;
  margin: 0 auto;
  display: block;
`

export default Img
