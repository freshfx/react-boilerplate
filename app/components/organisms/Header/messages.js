/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import {defineMessages} from 'react-intl'

const scope = 'boilerplate.organisms.Header'

export default defineMessages({
  features: {
    defaultMessage: 'Features',
    id: `${scope}.features`
  },
  home: {
    defaultMessage: 'Home',
    id: `${scope}.home`
  }
})
