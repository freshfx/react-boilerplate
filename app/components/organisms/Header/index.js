import React from 'react'
import {FormattedMessage} from 'react-intl'

import Anchor from './Anchor'
import Img from './Img'
import NavBar from './NavBar'
import HeaderLink from './HeaderLink'
import Banner from './banner.jpg'
import messages from './messages'

const Header = () => (
  <div>
    <Anchor href="https://twitter.com/mxstbr">
      <Img src={Banner} alt="react-boilerplate - Logo" />
    </Anchor>
    <NavBar>
      <HeaderLink to="/">
        <FormattedMessage {...messages.home} />
      </HeaderLink>
      <HeaderLink to="/features">
        <FormattedMessage {...messages.features} />
      </HeaderLink>
    </NavBar>
  </div>
)

export default Header
