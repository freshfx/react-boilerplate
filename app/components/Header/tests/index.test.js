import React from 'react'
import {render} from '@testing-library/react'
import {IntlProvider} from 'react-intl'

import Header from '../index'

jest.mock('react-router-dom', () => ({
  Link(props) {
    return <a {...props} href={props.to} />
  }
}))

const renderComponent = () =>
  render(
    <IntlProvider locale="en">
      <Header />
    </IntlProvider>
  )

describe('Header', () => {
  it('should match snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })
})
