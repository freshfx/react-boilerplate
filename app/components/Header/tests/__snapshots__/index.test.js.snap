// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
.c0 {
  color: #41addd;
  padding: 2em 0;
}

.c0:hover {
  color: #6cc0e5;
}

.c1 {
  width: 100%;
  margin: 0 auto;
  display: block;
}

.c2 {
  text-align: center;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0.25em 2em;
  margin: 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: bold;
  font-size: 16px;
  border: 2px solid #41addd;
  color: #41addd;
}

.c3:active {
  background: #41addd;
  color: #fff;
}

<div>
  <a
    class="c0"
    href="https://twitter.com/mxstbr"
  >
    <img
      alt="react-boilerplate - Logo"
      class="c1"
      src="IMAGE_MOCK"
    />
  </a>
  <div
    class="c2"
  >
    <a
      class="c3"
      href="/"
      to="/"
    >
      Home
    </a>
    <a
      class="c3"
      href="/features"
      to="/features"
    >
      Features
    </a>
  </div>
</div>
`;
